### PR TITLE
Update cosmos client to use v3 describe response

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: dcos/cosmos.py [hunk 1]
```diff
@@ -45,8 +45,8 @@ def _data(versions, http_method):
         self._endpoint_data = {
             'capabilities': _data(['v1'], 'get'),
             'package/add': _data(['v1'], 'post'),
-            'package/describe': _data(['v2', 'v1'], 'post'),
-            'package/install': _data(['v2', 'v1'], 'post'),
```
Comment (cruhland) [inline dcos/cosmos.py]: It's been long enough that we can drop support for the older versions?

Comment (jsancio) [inline dcos/cosmos.py]: This version of the CLI can only work against DC/OS 1.10. cc @tamarrow

**Comment by ssk2** [comment]
I tried to run this version locally (using the built version in TeamCity) and got this error:
```sh
➜  Downloads ./dcos package install hello-world
Item header 'Accept' deemed invalid by rule: Media type was [application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3] but should be one of [application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v2, application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v1]
```

**Comment by ssk2** [comment]
I still cannot get this working against testing/master:
```
➜  Downloads ./dcos package install --yes hello-world
Traceback (most recent call last):
  File "cli/dcoscli/subcommand.py", line 103, in run_and_capture
  File "cli/dcoscli/package/main.py", line 22, in main
  File "cli/dcoscli/util.py", line 24, in wrapper
  File "cli/dcoscli/package/main.py", line 36, in _main
  File "dcos/cmds.py", line 43, in execute
  File "cli/dcoscli/package/main.py", line 347, in _install
  File "dcos/packagemanager.py", line 447, in options
  File "dcos/packagemanager.py", line 431, in marathon_json
AttributeError: 'CosmosPackageVersion' object has no attribute 'package_version'
```

**Comment by ssk2** [comment]
Note that I appear to be unable to uninstall packages with this version of the CLI:
```
➜  cli git:(a323b42) ✗ dcos package uninstall hello-world
Traceback (most recent call last):
  File "/Users/sunil/workspace/dcos-cli/cli/dcoscli/subcommand.py", line 103, in run_and_capture
    exit_code = m.main([self._command] + self._args)
  File "/Users/sunil/workspace/dcos-cli/cli/dcoscli/package/main.py", line 22, in main
    return _main(argv)
  File "/Users/sunil/workspace/dcos-cli/cli/dcoscli/util.py", line 24, in wrapper
    result = func(*args, **kwargs)
  File "/Users/sunil/workspace/dcos-cli/cli/dcoscli/package/main.py", line 36, in _main
    return cmds.execute(_cmds(), args)
  File "/Users/sunil/workspace/dcos-cli/dcos/cmds.py", line 43, in execute
    return function(*params)
  File "/Users/sunil/workspace/dcos-cli/cli/dcoscli/package/main.py", line 485, in _uninstall
    package_manager, package_name, remove_all, app_id, cli, app)
  File "/Users/sunil/workspace/dcos-cli/dcos/package.py", line 32, in uninstall
    pkg, app_id, package_name, cli_only=False)
  File "/Users/sunil/workspace/dcos-cli/dcos/package.py", line 118, in installed_packages
    return merge_installed(apps, subcommands, app_id is not None, cli_only)
  File "/Users/sunil/workspace/dcos-cli/dcos/package.py", line 159, in merge_installed
    indexed_subcommands = {subcmd['name']: subcmd for subcmd in subcommands}
  File "/Users/sunil/workspace/dcos-cli/dcos/package.py", line 159, in <dictcomp>
    indexed_subcommands = {subcmd['name']: subcmd for subcmd in subcommands}
KeyError: 'name'```

File: dcos/cosmos.py (hunk 1 above)
Comment (jsancio) [inline dcos/cosmos.py]: @klueska After this change this version of the CLI will only work against 1.10 clusters.

File: dcos/packagemanager.py [hunk 2]
```diff
@@ -428,7 +428,7 @@ def marathon_json(self, options):
         """
 
         params = {"packageName": self.name()}
-        params["packageVersion"] = self.package_version()
+        params["packageVersion"] = self.version()
```
Comment (klueska) [inline dcos/packagemanager.py]: It seems weird to me to have the previous line set `packageName` inside the dictionaty definition, and then set this one outside of it. Can we change this to:
```
-        params = {"packageName": self.name()}
-        params["packageVersion"] = self.package_version()
+        params = {
+            "packageName": self.name(),
+            "packageVersion": self.version()
+        }
```

**Comment by klueska** [comment]
I've got all of the changes in this PR merged in my PR branch (which is finally passing all the tests). Therefore, I'm going to manually bypass the checks here and merge this PR manually to close it out.

**Pull request merged**